### PR TITLE
Snapshot name should not contain special characters

### PR DESCRIPTION
**Pull request by sarankk**

### CODE REVIEW

File: test/unit/org/apache/cassandra/tools/nodetool/SnapshotTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import org.apache.cassandra.io.util.File;
+import org.apache.cassandra.tools.ToolRunner;
+
+import static org.apache.cassandra.tools.ToolRunner.invokeNodetool;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**o
+ * Tests for the {@code nodetool snapshot} command
+ */
+public class SnapshotTest extends CQLTester
```
Comment (skoppu22) [inline test/unit/org/apache/cassandra/tools/nodetool/SnapshotTest.java:34]: Awesome, nice to see extensive test cases

**Comment by skoppu22** [comment]
> _No description provided._

Don't we have jira ticket for this ?

**Comment by smiklosovic** [comment]
Yes, please create a ticket.

File: src/java/org/apache/cassandra/tools/nodetool/Snapshot.java [hunk 2]
```diff
@@ -73,6 +74,10 @@ public void execute(NodeProbe probe)
                 options.put("ttl", d.toString());
             }
 
+            if (!snapshotName.isEmpty() && snapshotName.contains(File.pathSeparator()))
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/tools/nodetool/Snapshot.java:77]: is path separator the only character illegal to be included in a snapshot name?

Comment (skoppu22) [reply src/java/org/apache/cassandra/tools/nodetool/Snapshot.java:77]: What about null character "\0" ?

Comment (smiklosovic) [reply src/java/org/apache/cassandra/tools/nodetool/Snapshot.java:77]: @skoppu22 I would like to have that discussion in the ticket. Ticket is the must anyway in order to merge it. No way around it.

Comment (skoppu22) [reply src/java/org/apache/cassandra/tools/nodetool/Snapshot.java:77]: @smiklosovic I am not the author of the PR

**Comment by smiklosovic** [comment]
hi @sarankk , could you create a ticket in jira.apache.org for Cassandra project? thank you.

**Comment by smiklosovic** [comment]
done in 15297

**Pull request closed without merging**